// ngx-datetimepicker/ngx-datetimepicker/src/app/dateTimePicker.component/dateTimePicker.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter, ViewEncapsulation, HostListener, ElementRef } from '@angular/core';
import { IsMobileService } from '../services/isMobile.service';
import { DateService } from '../services/date.service';
export var DateTimePickerComponent = (function () {
    function DateTimePickerComponent(isMobileService, dateService, eRef) {
        this.isMobileService = isMobileService;
        this.dateService = dateService;
        this.eRef = eRef;
        this.selectedDateTimeChange = new EventEmitter();
        this.pickerVisible = false;
        this.isMobile = isMobileService.isMobile;
    }
    DateTimePickerComponent.prototype.offClick = function (event) {
        if (!this.eRef.nativeElement.contains(event.target)) {
            this.pickerVisible = false;
        }
    };
    Object.defineProperty(DateTimePickerComponent.prototype, "formattedDate", {
        get: function () {
            return this.dateService.formatMMDDYYYY_HHMM_AMPM(this.selectedDateTime);
        },
        set: function (value) {
            this.selectedDateTime = new Date(value);
        },
        enumerable: true,
        configurable: true
    });
    DateTimePickerComponent.prototype.ngOnInit = function () {
        //If no date is selected then default to today's date.
        if (!this.selectedDateTime) {
            this.selectedDateTime = new Date();
            this.selectedDateTime.setHours(12, 0);
        }
        if (typeof this.selectedDateTime == 'string') {
            this.selectedDateTime = new Date(this.selectedDateTime);
        }
    };
    DateTimePickerComponent.prototype.newDatePicked = function (date) {
        this.selectedDateTimeChange.emit(date);
        this.selectedDateTime = date;
    };
    DateTimePickerComponent.prototype.closePicker = function (close) {
        this.pickerVisible = close;
    };
    __decorate([
        Input(), 
        __metadata('design:type', Date)
    ], DateTimePickerComponent.prototype, "selectedDateTime", void 0);
    __decorate([
        Output(), 
        __metadata('design:type', Object)
    ], DateTimePickerComponent.prototype, "selectedDateTimeChange", void 0);
    __decorate([
        HostListener('document:click', ['$event']), 
        __metadata('design:type', Function), 
        __metadata('design:paramtypes', [Object]), 
        __metadata('design:returntype', void 0)
    ], DateTimePickerComponent.prototype, "offClick", null);
    DateTimePickerComponent = __decorate([
        Component({
            selector: 'ngx-datetime-picker',
            templateUrl: './datetimepicker.component.html',
            encapsulation: ViewEncapsulation.None,
            styleUrls: ['../../assets/date-picker.css']
        }), 
        __metadata('design:paramtypes', [IsMobileService, DateService, ElementRef])
    ], DateTimePickerComponent);
    return DateTimePickerComponent;
}());
